lib/msprime.h

```diff
@@ -129,6 +129,7 @@ typedef struct {
         double *allele_frequency;
         double *time;
         size_t num_steps;
+        double sweep_dt;
```

Comment:
Is this needed as a separate parameter (i.e., it's not the difference between adjacent values in the time array)? If so, it would be good to add a comment here explaining what it's for, as it's not obvious.

Reply:
probably not needed. i put this in to give us options design-wise for the trajectory times. we can definitely do it from the difference between adjacent times.

Reply:
OK, let's change it over to computing the dts online. I want to keep things simple as possible at the C level; we can add user friendly stuff in Python. Would you mind updating to remove this? I think we can merge then, and I'll setup the Python-C plumbing. I see what you mean about the absolute times too, BTW, we should probably change these to relative to the start of the sweep.

Reply:
happy to change over the dt parameter today. there is a big obstacle to changing over times to be relative to the beginning of the sweep: we need to account for population size changes when we construct the trajectories themselves. 

i think we should keep them absolute for the time being and start to nail down how we will implement the trajectory generation. For instance should I port over C code to do this? Should we instead do the trajectories using python?

Reply:
OK, understood and agreed on the times. I want to do as little as possible in C - I'm sure we can use the DemographyDebugger to give us the population sizes that we need, and numpy arrays used well should be fast enough. If you can get rid of the dts I'll make the low-level C-python interface and we'll be able to see where to go from there.

Reply:
okay took at the `sweep_dt` thing and moved to calculating those on the fly.